client/src/pages/profile/index.tsx

```diff
@@ -107,210 +91,180 @@ export class ProfilePage extends React.Component<Props, State> {
-      this.setState({ isSaving: false });
+      setIsSaving(false);
       message.success('Profile was successfully saved');
       isUpdated = true;
     } catch (error) {
-      this.setState({ isSaving: false });
+      setIsSaving(false);
       message.error('Error has occurred. Please check your connection and try again');
       isUpdated = false;
     }
 
     return isUpdated;
   };
 
-  authorizeDiscord = async () => {
-    this.setState({ isLoading: true });
-
-    const discord = await this.userService.getDiscordIds();
-
+  const authorizeDiscord = async () => {
+    const discord = await userService.getDiscordIds();
     if (discord) {
-      this.setState(({ profile, ...state }) => ({
-        ...state,
-        profile: {
-          ...profile,
-          publicCvUrl: profile?.publicCvUrl ?? null,
-          discord,
-        },
+      setProfile(profile => ({
+        ...profile,
+        publicCvUrl: profile?.publicCvUrl ?? null,
+        discord,
       }));
 
-      await this.updateProfile({ discord });
-      this.props.router.replace('/profile');
+      await updateProfile({ discord });
+      router.replace('/profile');
     }
-
-    this.setState({ isLoading: false });
   };
 
-  componentDidMount() {
-    // it's a dirty hack to fix an issue with empty query params
-    // see: https://nextjs.org/docs/routing/dynamic-routes#caveats
-    //
-    // >> After hydration, Next.js will trigger an update to your application
-    // >> to provide the route parameters in the query object.
-    setTimeout(async () => {
-      await this.fetchData();
-      await this.authorizeDiscord();
-    }, 100);
-  }
-
-  render() {
-    const { profile, isProfileOwner, connections } = this.state;
-
-    const mainInfo: ProfileMainCardData = {
-      location: profile?.generalInfo?.location ?? null,
-      name: profile?.generalInfo?.name ?? '',
-      githubId: profile?.generalInfo?.githubId ?? null,
-      publicCvUrl: profile?.publicCvUrl ?? null,
-    };
-    const aboutMyself = profile?.generalInfo?.aboutMyself ?? '';
-    const languages = profile?.generalInfo?.languages ?? [];
+  const mainInfo: ProfileMainCardData = {
+    location: profile?.generalInfo?.location ?? null,
+    name: profile?.generalInfo?.name ?? '',
+    githubId: profile?.generalInfo?.githubId ?? null,
+    publicCvUrl: profile?.publicCvUrl ?? null,
+  };
+  const aboutMyself = profile?.generalInfo?.aboutMyself ?? '';
+  const languages = profile?.generalInfo?.languages ?? [];
 
-    const githubId = profile?.generalInfo?.githubId;
-    const isAdmin = this.props.session.isAdmin;
+  const githubId = profile?.generalInfo?.githubId;
+  const isAdmin = session.isAdmin;
 
-    const cards = [
-      profile?.generalInfo && (
-        <MainCard data={mainInfo} isEditingModeEnabled={isProfileOwner} updateProfile={this.updateProfile} />
-      ),
-      <AboutCard
-        key="about-card"
-        data={aboutMyself}
+  const cards: JSX.Element[] = [
```

Comment:
you not only setting type here, but also casting it again at the end) 
though ts would know the type even if you don't set it at all.